src-react/components/AddonTable.jsx

```diff
@@ -60,34 +61,58 @@ export class AddonTable extends Component {
     this.setState({ addonList })
   }
 
+  // Sends request to install addon
   onInstall (addonObj) {
     ipcRenderer.send('installAddon', addonObj)
   }
 
-  onUpdate (addonObj) {
-    ipcRenderer.send('updateObj', addonObj)
+  // Sends request to check addon for updates
+  onCheckUpdate (addonObj) {
+    ipcRenderer.send('checkAddonUpdate', addonObj)
   }
 
-  onRemove () {
-    // noop
+  // Sends request to install an update
+  onInstallUpdate (addonObj) {
+    ipcRenderer.send('installUpdate', addonObj)
+  }
+
+  onRemove (addonObj) {
+    ipcRenderer.send('uninstallAddon', addonObj)
   }
 
   renderRow (addonObj, key) {
-    const btnTag = addonObj.dlStatus === 100
-      ? <button onClick={() => this.onUpdate(addonObj)}>Update</button>
-      : <button onClick={() => this.onInstall(addonObj)}>Install</button>
+    let btnTag
+    let dlStatus
+    switch (addonObj.status) {
+      case 'INSTALLED':
+        btnTag = <button onClick={() => this.onCheckUpdate(addonObj)}>Check Update</button>
+        break;
+      case 'NEW UPDATE':
+        btnTag = <button onClick={() => this.onInstallUpdate(addonObj)}>Update</button>
+        break;
+      default:
+        btnTag = <button onClick={() => this.onInstall(addonObj)}>Install</button>
+    }
+
+    if (addonObj.dlStatus !== 100) {
+      dlStatus = <td>{addonObj.dlStatus}</td>
+      if ( (addonObj.dlStatus >= 0) && (addonObj.dlStatus < 100) ){ // Currently downloading something so hide the install button
+        btnTag = ""
+      }
+    }
 
     return (
       <tr key={key}>
-        <td>{addonObj.displayName}</td>
         <td>{addonObj.host}</td>
+        <td><a href="#" onClick = {() => shell.openExternal(addonObj.URL)}>{addonObj.displayName}</a></td>
```

Comment:
can we just have a tool tip that displays the url? I'd perfer to use `shell.openExternel` as little as possible. it seems like bad practice and I was using it as a place holder for the issues button.

Reply:
Yea whatever is better. Maybe we can also put the authors names in the tooltip too since there doesn't seem to be a condense way to list them in the table.

Reply:
The authors tooltip is something I have on the backburner right now